inc/vcf/fixer.hpp

```diff
@@ -117,33 +118,20 @@ namespace ebi
             // TODO better log system, if any
             std::cerr << "DEBUG: line " << error.get_line() << ": fixing invalid INFO field " << error.get_field()
                       << std::endl;
-            const int info_column_index = 7;
+            const size_t info_column_index = 7;
             const std::string empty_info_column = ".";
 
             size_t num_removed_subfields = 0;
             std::string string_line = {line->begin(), line->end()};
 
-            fix_column(info_column_index, string_line, "\t", [&](std::string &info_column) {
-                std::vector<std::string> info_subfields, keys_values;
-                util::string_split(info_column, ";", info_subfields);
-
-                for (size_t j = 0; j < info_subfields.size(); ++j) {
-                    util::string_split(info_subfields[j], "=", keys_values);
-                    if (keys_values[0] != error.get_field()) {
-                        if (num_removed_subfields != j) {
-                            // if this is not the first field we don't remove, then write the separator
-                            output << ";";
-                        }
-                        output << info_subfields[j];
-                    } else {
-                        num_removed_subfields++;
-                    }
-                }
+            auto remove_info_field_if = [&](std::string &info_subfield, size_t index) -> bool {
```

Comment:
Could this be renamed to `condition_to_remove_info_field`? I didn't understand what it did until I saw the `remove_column` definition.